### DISCUSSION TITLE
[Question][Hw7] Questions about hw7

### DISCUSSION THREAD

**Discussion by h2oche** [prosyslab-classroom/cs524-program-analysis#386] [Q&A]
Hello, I have some questions about homework 7. 

- Is id number important in grading? When I run `./typechecker test/example1.ll`, all results seems to be same with `test/example1.expected` except for id numbers (ex: ID XX). For that reason, when I run a test script, it fails. I guess it depends on how we add constraints to solver, and it may vary.
- Is a function pointer type not a scope of homework 7? When I tried to give a type to `%fp = alloca void (i32)*, align 8`, it throws an exception(with message `"unsupported type"`) in `type_of_lltype`.

Thanks!

**Comment by kangwoosukeq** [comment]
- Our current scoring script compares only whether the results match. But even if the results are not the same, we will give points if your code is correct.

- The function pointer is not the scope of this homework. You don't have to care about that.
